### ISSUE TITLE
basic.cancel event?

### ISSUE THREAD

**Issue by paul-bell** [cloudamqp/amqp-client.js#24]
Hello, 

Does your Node.js client allow handling of basic.cancel event as described here: https://www.rabbitmq.com/consumer-cancel.html ?

If so, would you be so kind as to provide an example of how to do this? I am quite interested in using your client, but could not find such an example in the tests directory.

Thank you.

**Comment by carlhoerberg** [comment]
Yes, of course! when subscribing to a queue you get an `AMQPConsumer`, back, which got a [`cancel`](https://cloudamqp.github.io/amqp-client.js/classes/AMQPConsumer.html#cancel) method:

```js
    const consumer = await q.subscribe({noAck: true}, async (msg) => {
      console.log(msg.bodyToString())
      await consumer.cancel()
    })
   await consumer.wait(10000) // wait maximum 10s for the consumer to quit
```

But you also got the raw [`AMQPChannel#basicCancel`](https://cloudamqp.github.io/amqp-client.js/classes/AMQPChannel.html#basicCancel) method:

```js
const amqp = new AMQPClient("amqp://localhost")
const conn = await amqp.connect()
const ch = await conn.channel()
const consumerTag = await ch.basicConsume("any-existing-queue", (msg) => { work(msg) })
await ch.basicCancel(consumerTag)
```

**Comment by paul-bell** [comment]
Thank you for this very prompt reply; much appreciated!

Perhaps I am confused, but I am not interested in a method that cancels. Rather, I am interested in receiving a 'client notification' in the form of a 'basic.cancel' _event_ sent by the broker. Are we talking about the same thing from different perspectives?

**Comment by carlhoerberg** [comment]
Ok, now I'm with you! And correct, that wasn't supported until now. Let me know if you think that API is good enough.

**Comment by carlhoerberg** [comment]
Server sent cancel is now supported and included in v2.0.0, happy to hear any feedback.
